finagle-http/src/main/scala/com/twitter/finagle/Http.scala

```diff
@@ -268,6 +269,13 @@ object Http extends Client[Request, Response] with HttpRichClient
     def withCompressionLevel(level: Int): Client =
       configured(param.CompressionLevel(level))
 
+
+    /**
+     * Enable the collection of HTTP specific metrics. See [[http.filter.StatsFilter]].
+     */
+    def withHttpStats: Client =
```

Comment:
lets add a basic unit test that verifies these work as expected. something like a request populates at least one of the stats. same for the server side.